src/txn/txn_ckpt.c

```diff
@@ -1487,6 +1487,16 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
     if (tracking)
         WT_TRET(__wt_meta_track_off(session, false, failed));
 
+    /*
+     * Update the global checkpoint ID in disaggregated storage. This has to be done after
+     * checkpoint resolve, which happens when we turn off metadata tracking above.
+     */
+    if (!failed && ret == 0 /* ensure that turning off meta tracking worked */) {
+        WT_ACQUIRE_READ(num_meta_put, conn->disaggregated_storage.num_meta_put);
+        if (conn->disaggregated_storage.num_meta_put_at_ckpt_begin < num_meta_put)
+            WT_TRET(__wt_disagg_advance_checkpoint(session));
+    }
+
```

Comment:
I've been thinking that disagg checkpoints are totally independent of traditional WT checkpoints, but this indicates that's not true.  I think I'll need to understand more about how/why they have somewhat synchronized activities here.

Reply:
WT checkpoints make a logical sense because they force the internal pages all the way round the root to be written, and then the metadata to be written, so new data are accessible. The original design goal was that once you complete a checkpoint, the follower can pick up the new checkpoint in its entirety, with all data accessible, which requires all of these internal pages and metadata to be written in the checkpoint.

Furthermore, they have to be tied closely together because we use only checkpoint IDs for their versioning - so we are only allowed to reference the latest page version within the SLS checkpoint. If, let's say, we evict a page without updating the internal page, we would have data inconsistency.

Reply:
Thanks for the explanation!